src/validators/models.rs

```diff
@@ -0,0 +1,201 @@
+            }
+            return Err(ValidationError::Unknown { 
+                code: "action_not_found".into(),
+                detail: Some(format!("Could not find action: {}", self.uses)),
+                path: self.origin.to_owned(),
+                title: "Action Not Found".into(),
+            });
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct Docker {
+    pub uses: String,
+    pub origin: String,
+    pub image: String,
+    pub url: Option<String>,
+    pub tag: Option<String>,
+}
+
+impl Docker {
+    pub const PATTERN: &str = r"(?P<Docker>.{0}(?:docker://)(?P<url>([^/:]+)\.([^/:]+)/)?(?P<image>[^:]+)(?::(?P<tag>.+))?)";
+}
+
+impl Uses<'_> for Docker {
+    #[cfg(not(feature="remote-checks"))]
+    fn validate(&self) -> Result<(), ValidationError> {
+        Ok(())
+    }
+
+    /// Remote-check validation logic for Docker images. If the destructured image is able to be
+    /// retrieved from Docker Hub or using the registries v2 endpoints, then the image exists. If
+    /// the image results in a 401 (UNAUTHORIZED), the image _could_ exist. In this case,
+    /// action-validator will assume the image exists.
+    #[cfg(feature="remote-checks")]
+    fn validate(&self) -> Result<(), ValidationError> {
+        let mut image = self.image.to_owned();
+
+        // If there's no prefix and the image is in docker hub, then it means the image is a
+        // library image.
+        if self.url.is_none() && !image.contains("/") { 
+            image = format!("library/{image}");
+        }
+
+        let url = match (self.url.as_ref(), self.tag.as_ref()) {
+            // Lookup V2 protocol tag
+            (Some(url), Some(tag)) => format!("https://{url}/v2/{image}/manifests/{tag}"),
+            // Lookup V2 protocol image
+            (Some(url), None) => format!("https://{url}/v2/{image}/tags/list"),
+            // Lookup docker hub tag
+            (None, Some(tag)) => {
+                format!("https://registry.hub.docker.com/v2/repositories/{image}/tags/{tag}")
+            },
+            // Lookup docker hub image
+            (None, None) => {
+                format!("https://registry.hub.docker.com/v2/repositories/{image}")
+            },
+        };
+
+        if let Some(r) = _get_request(url).ok() {
+            let status = r.status();
+            if status == StatusCode::OK {
+                return Ok(());
+            } else if status == StatusCode::UNAUTHORIZED {
+                // In this case, pull access requires authorization. There are simple cases that
+                // only require the bearer token retrieval followed by manifest access, but there
+                // are also others that require user credentials. For now, we should assume that
+                // the tag tag is valid. We could perform authentication, but that would requrie
+                // user creds and adds a whole lot of scope to this feature.
+                return Ok(());
```

Comment:
I'm OK with breaking API changes when we're early in the lifecycle, and I don't have strong opinions on either of these approaches, but it does seem like a good idea to be recording problems for later consumption, rather than logging anything immediately.